core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -1234,74 +1251,28 @@ public int compare(FingerprintFacet o1, FingerprintFacet o2) {
     }
 
     /**
-     * Save the settings to a file.
+     * Save the Fingerprint in the Fingerprint Storage
      * @throws IOException Save error
      */
     public synchronized void save() throws IOException {
-        if(BulkChange.contains(this))   return;
+        if(BulkChange.contains(this)) return;
 
         long start=0;
         if(logger.isLoggable(Level.FINE))
             start = System.currentTimeMillis();
 
-        File file = getFingerprintFile(md5sum);
-        save(file);
-        SaveableListener.fireOnChange(this, getConfigFile(file));
+        FingerprintStorage.get().save(this);
 
         if(logger.isLoggable(Level.FINE))
-            logger.fine("Saving fingerprint "+file+" took "+(System.currentTimeMillis()-start)+"ms");
+            logger.fine("Saving fingerprint took "+(System.currentTimeMillis()-start)+"ms");
```

Comment:
It still needs some kind of fingerprint identifier in the message, e.g. just its ID/hash